### DISCUSSION TITLE
WIP Release notes (v5.9)

### DISCUSSION THREAD

**Discussion by Thanatomanic** [Beep6581/RawTherapee#6331] [General]
## This is a collection of information related to the future release of RawTherapee 5.9. This is a work in progress and therefore bound to be incomplete!

Based on the merged PR's, there have been valuable contributions by (in alphabetical order) @10110111, @Benitoite, @Bezierr, @CarVac, @Desmis, @EwoutH, @Floessie, @Lawrence37, @Pandagrapher, @PureW, @SimonSegerBlomRex, @Thanatomanic, @adamreichold, @dheijl, @ff2000, @heckflosse, @jonathanBieler, @kdm9, @luzpaz, @mkyral, @pinchartl, @rfranke, @rom9, @stwunsch, @ya-isakov, @ylep and two honorable mentions for @freddii and @shapirus who both added one single line to the codebase.

As per commit https://github.com/Beep6581/RawTherapee/commit/b1e7860a234109761fb9e693e0c603cf0c58827f there have been **114,120 insertions** and **20,714 deletions** in the code since the 5.8 release.

# Noteworthy new features and improvements

### Introducing local adjustments
Main PR: #5755
Auxilliary work: #5727, #5731, #5743, #5744, #5750, #5911, #5776
Other minor fixes: #5916, #5914, #5918, #5920, #5921, #5924, #5927, #5925, #5932, #6065, #6058, #5989, #5937, #5944, #5941, #5981, #6093, #6120, #6149, #6172, #6179, #6091, #6160, #6199, #6228, #6238, #6265, #6277, #6313
Common color mask (#5810)
Improved exposure with Laplacian PDE (#5832)
Introduce complexity modes (basic, normal, expert) (#5858)
Improved highlight compression (#5820)
Introduce C=f(H) curve in Color and Light (#5847)
Introduce log encoding using CIECAM02 (#5973)
Introduce grey area denoising (#6037)
Log encoding - Recovery based on mask luminance (#6038)
Shadows highlights - Recovery based on luminance mask #6050
denoise - Change curve denoise and default settings #6051
Vibrance and Warm Cool - Recovery based on luminance mask #6055
Color and light - Recovery based on luminance mask #6035
Denoise and Guided Filter - Recovery based on luminance mask #6032
added Laplacian to detail threshold DCT #6027
Introduce Full Image mode (#6063)
Retinex - Last Recovery based on luminance mask #6062
CBDL Luminance - Recovery based on luminance mask #6060
Tone Mapping - Recovery based on luminance mask #6059
local contrast - wavelet - Recovery based on luminance mask #6057
Changes exposure response with tonecurve and Laplacian #6009
Various improvment Log encoding - Basic - Standard - Advanced (better integration with CIECAM, added mask, levels of complexity) #5979
Mask - Change behavior Laplacian #6075
Local adjustments are placed in the pipeline very early, before all RGB-based tools (#6080)
deltaE preview does not trigger further image processing, not influencing the preview (#6099)
Blur and noise radius > 1.5 increase saturation issue #6117 #6118
Contrast modulation in Log encoding / CIECAM16 (#6169)
Normalize luminance - improvment Preview TM - Retinex - Remove D.Range #6185
Strength sliders to various tools (#6223)

### Generalized camera-based perspective correction (#5569)
Improve control line behavior #6268

### Improved inspector window (#5593, #5872, #6222)
Can now open preview of image from file browser (f key). Can zoom and pan. Configurable from Preferences, old inspector still functional.

### Some changes in processing pipeline.
Profiled distortion correction before auto-fill and perspective correction (#5595)
CHECK! Local contrast is now earlier in the pipeline to avoid clash with tone-mapping (#6134)
CHECK! Shadows and highlights moved in pipeline (#6164)
CHECK! Move colortoning labgrid from rgb to lab (#6173)

### Film negative module
Calculate channel multipliers from the current crop area (#5485)
Film negative mode now also works on non-raw files, and with better accuracy (#5798)

### Spot removal tool (#6227)
Ability to remove blemishes.

### Abstract profiles (#5949)
Allows new type of manipulation of images, by simulating other TRC's, primaries and white points.
With interactive CIExy diagram to choose primaries and white point (#6207)
Some minor changes #6243

### New default theme (#5929)
Some styling changes. Particular important change is a clearly distinctive icon that emphasizes deactivated and activated modules.

### Preprocess white balancing (#5662)
Choice to keep the camera raw multipliers before demosaicing or do automatic grey-world (?) white balancing before demosaicing. This can sometimes help reduce color artifacts (for examples, see #5616 and #5662).

### New blur mode for color propagation highlight recovery (#6109)
Helps getting rid of harsh transitions

### Bilinear demosaicing (#5885)
Dual mosaicing can now use a simple bilinear demosaicing for flat areas without significant loss of quality in those regions over other, slower demosaicers.

### Various improvements to Wavelet
Blur levels and Residual image introduced (#5694)
Threshold in contrast levels (#5695)
Other modifications (#5700)
Chrominance denoising (#5702)
Change sensitivity of sliders for contrast levels (#5705)
Various changes to clarity / sharp mask / etc (???) (#5660)
Fixes for sliders instead of curve (#5795)
Complexity modes (standard, advanced) (#5869)
In (#5894)
    "denoise luminance" : addition of a hue curve, and take into account "local contrast" to differentiate the action between solids and structures
    "final touchup" : addition of "Guided filter" with 2 features added : curve Hue, and "Local contrast"
Denoise disables contrast levels above level 6 (#5942)

### Various improvements to Retinex
Introduce complexity modes (standard, advanced) (#5871)

### Various improvements for compilation and bundling on macOS, including native compilation on M1 hardware
#5644, #5641, #5674, #5703, #5771, #6004, #6152, #5786

### Various speedups and code quality improvements
#5610, #5611, #5617, #5648, #5669, #5675, #5684, #5803, #5815, #5816, #5884, #5878, #5902, #6047, #6064, #5996, #5866, #5850, #5836, #5846, #5781, #6113, #6122, #6126, #6128, #6135, #6133, #6145, #6138

### Various fixes
#5539, #5613, #5639, #5671, #5685, #5742, #5739, #5765, #5797, #5822, #5823, #5834, #5808, #5896, #5854, #5939, #5997, #5930, #5948, #5966, #6141, #6147, #6156, #6158, #6127, #6190, #6253, #6278, #6305, #6317

### Explicit camera support or improved color rendition (both for older and newer models)
Note: there are more camera's supported than listed here, in some cases even future ones. An exhaustive list is not available, but let us know if your camera is not supported yet.
Canon EOS 100D / Rebel SL1 / Kiss X7 (#6187)
Canon EOS 2000D / Rebel T7 / Kiss X90
Canon EOS 400D DIGITAL
Canon EOS 5D Mark IV (DCP)
Canon EOS 90D (DCP)
Canon EOS M6 Mark II (DCP)
Canon EOS R (DCP)
Canon EOS R5 (#5905) (DCP)
Canon EOS R6 (#5905) (DCP)
Canon EOS RP (#6187)
Canon EOS-1D Mark III
Canon EOS-1Ds
Canon PowerShot G1 X Mark II (DCP)
Canon PowerShot G9 X Mark II
Canon PowerShot S120 (DCP)
Canon PowerShot SX50 HS
Canon PowerShot SX70 HS
DJI FC3170
FUJIFILM X-A5 (DCP)
FUJIFILM X-E4
FUJIFILM X-H1 (DCP)
FUJIFILM X-PRO2
FUJIFILM X-PRO3 (DCP)
FUJIFILM X-S10
FUJIFILM X-T1
FUJIFILM X-T100
FUJIFILM X-T2
FUJIFILM X-T3 (DCP)
FUJIFILM X-T30
FUJIFILM X-T4
FUJIFILM X100V
Fujifilm GFX 100 (#6208)
Fujifilm GFX100S (#6208) * Note that lossy compression is not supported, nor are alternative crop modes (e.g. 4:3)
Fujifilm X-A20
HASSELBLAD NEX-7 (Lunar)
Hasselblad L1D-20c (DJI Mavic 2 Pro)
LEICA C-LUX
LEICA CAM-DC25
LEICA D-LUX 7
LEICA M8
LEICA V-LUX 5
Leica SL2-S
NIKON COOLPIX P1000
Nikon 1 J4
Nikon COOLPIX P950
Nikon D2Hs
Nikon D2Xs
Nikon D300s
Nikon D3500
NIKON D500 (DCP)
NIKON D5300 (DCP)
Nikon D6
NIKON D610 (DCP)
Nikon D70s
NIKON D7100 (DCP)
NIKON D7500 (DCP)
Nikon D780
NIKON D800 (DCP)
Nikon D810A
NIKON D850 (DCP)
Nikon Z 5 (#5905)
Nikon Z 50 (#5851) (DCP)
NIKON Z 6 (DCP)
Nikon Z 6_2
NIKON Z 7 (DCP)
Nikon Z 7_2
OLYMPUS E-M10MarkIV
OLYMPUS E-M1MarkIII
OLYMPUS E-M1X
OLYMPUS E-M5MarkII (DCP)
OLYMPUS E-M5MarkIII
OLYMPUS E-PL10
OLYMPUS E-PL9
OLYMPUS STYLUS1
OLYMPUS STYLUS1,1s
OLYMPUS TG-6
PENTAX K10D
PENTAX K-50 (DCP)
Panasonic DC-FZ1000M2
Panasonic DC-FZ80
Panasonic DC-FZ81
Panasonic DC-FZ82
Panasonic DC-FZ83
Panasonic DC-G100
Panasonic DC-G110
Panasonic DC-G90
Panasonic DC-G95
Panasonic DC-G99
Panasonic DC-S1H
Panasonic DC-S5 (DCP)
Panasonic DC-TZ95
Panasonic DC-ZS80
Panasonic DMC-TZ80
Panasonic DMC-TZ85
Panasonic DMC-ZS60
RICOH PENTAX K-1 Mark II
RICOH PENTAX K-3 MARK III
SONY NEX-7
Sigma fp
Sony DCZV1B
Sony DSC-HX95
Sony DSC-HX99
Sony DSC-RX0
Sony DSC-RX0M2
Sony DSC-RX100
Sony DSC-RX100M5A
Sony DSC-RX100M6
Sony DSC-RX100M7
Sony DSC-RX10M2
Sony DSC-RX10M3
Sony DSC-RX10M4
Sony DSC-RX1R
Sony ILCE-1
Sony ILCE-6100
Sony ILCE-6400 (DCP)
Sony ILCE-6600 (DCP)
Sony ILCE-7C
Sony ILCE-7M3
Sony ILCE-7RM4 (DCP)
Sony ILCE-7SM3
SONY ILCE-9 (DCP)
Sony ILCE-9M2
Sony NEX-F3
Sony SLT-A99V

**Comment by rom9** [comment]
> Calculate channel multipliers from the current crop area (#5485)

Sorry, the initial description of this PR is now a lie :-D
Multipliers are not computed from the crop area, that was a first attempt before moving the filmneg processing downstream of demosaic/WB. It has been superceded by the dedicated color balance sliders in the current version.

Maybe something like:
_"Channel multipliers are now saved to the processing profile, and can be copied for consistent processing of multiple frames (#5485)"_

would better describe the benefit added by that PR.
Do you think i should edit the PR description ? (it seems to be separate from the commit comment)

Regarding the calculation of multipliers from a rectangular area, there is a separate issue (#6178), but it can get quite complex (fine rotation, perspective distortion, etc.), so i don't think it can be ready for v5.9 :-(

**Reply by Thanatomanic** [reply]
@rom9 
> Do you think i should edit the PR description ? (it seems to be separate from the commit comment)

Nah, that's not necessary, certainly not retroactively. I had just included that PR because I encountered it, not because it needs to be mentioned :-)
The final wording will focus on the overhaul, and your proposed sentence for the processing profile is useful to add!
